### PR TITLE
Add boto3 asg

### PR DESCRIPTION
**Pull request by binocvlar**
### What does this PR do?
Adds a "clone" of the existing `boto_asg.py` module as `boto3_asg.py`. This new module contains some light modifications in order to circumvent calls to problematic boto functions.

I'm on the fence about whether it would be better to simply update `boto_asg.py` with the new functions (perhaps named like `get_config_boto3` instead of `get_config`). The currently implemented approach does make delegation of ownership of the new function a little clearer and cleaner looking at the callsite, IMO. ~~For the time being I haven't pruned all of the unnecessary code from `boto3_asg` as I'm not entirely sure we've seen the end of required modifications.~~

### What issues does this PR fix or reference?
IT-101

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/boto_asg.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def get_config(name, region=None, key=None, keyid=None, profile=None):
     retries = 30
     while True:
         try:
-            asg = [g for g in conn.get_all_groups(names=[name]) if g.name]
+            asg = [g for g in conn.get_all_groups(names=[name]) if all([g.name, g.vpc_zone_identifier])]
```
Comment (abrom) [inline salt/modules/boto_asg.py]: do we still want this here? Or can we roll is back out?

File: salt/modules/boto3_asg.py [hunk 2]
```diff
@@ -0,0 +1,946 @@
...
+
+def __init__(opts):
+    salt.utils.compat.pack_dunder(__name__)
+    if HAS_BOTO:
+        __utils__['boto3.assign_funcs'](
+            __name__, 'autoscaling',
+            get_conn_funcname='_get_conn_autoscaling_boto3')
+
+
+def exists(name, region=None, key=None, keyid=None, profile=None):
```
Comment (abrom) [inline salt/modules/boto3_asg.py]: should we be including these methods that haven't been updated here given they will almost certainly be broken still?  (ie using `boto.get_all_groups`)

File: salt/modules/boto3_asg.py [hunk 3]
```diff
@@ -0,0 +1,946 @@
...
+            if retries and e.code == 'Throttling':
+                log.debug('Throttled by AWS API, retrying in 5 seconds...')
+                time.sleep(5)
+                retries -= 1
+                continue
+            log.error(e)
+            return {}
+
+
+def create(name, launch_config_name, availability_zones, min_size, max_size,
```
Comment (abrom) [inline salt/modules/boto3_asg.py]: same here.. if we aren't upgrading it to use boto3, not sure it should be copied across.

File: salt/modules/boto3_asg.py (hunk 2 above)
Comment (binocvlar) [reply salt/modules/boto3_asg.py]: Yep agreed that they should be removed, but I was planning on verifying that we've patched all that needs to be patched before removing them.

File: salt/modules/boto_asg.py [hunk 4]
```diff
@@ -141,7 +141,7 @@ def get_config(name, region=None, key=None, keyid=None, profile=None):
     retries = 30
     while True:
         try:
-            asg = [g for g in conn.get_all_groups(names=[name]) if g.name]
+            asg = conn.get_all_groups(names=[name])
```
Comment (binocvlar) [inline salt/modules/boto_asg.py:144]: Note: this is a result of reverting the changes shown on the left :)

File: salt/modules/boto_asg.py [hunk 5]
```diff
@@ -781,7 +781,7 @@ def get_all_groups(region=None, key=None, keyid=None, profile=None):
             asgs = []
             while next_token is not None:
                 ret = conn.get_all_groups(next_token=next_token)
-                asgs += [a for a in ret if a.name]
+                asgs += [a for a in ret]
```
Comment (binocvlar) [inline salt/modules/boto_asg.py:785]: Note: this is a result of reverting the changes shown on the left :)

File: salt/modules/boto_asg.py (hunk 1 above)
Comment (binocvlar) [reply salt/modules/boto_asg.py]: Yep, both of those versions of that line (-/+) have been reverted back to the original.

File: salt/modules/boto3_asg.py (hunk 2 above)
Comment (binocvlar) [reply salt/modules/boto3_asg.py]: These have now been removed now that the initial modifications have been tested.

File: salt/modules/boto3_asg.py (hunk 3 above)
Comment (binocvlar) [reply salt/modules/boto3_asg.py]: See above.

**Review by abrom — APPROVED** [review]
LGTM

**Pull request merged**